marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/flow/getFlowWithLatestJobInfo.sjs

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+const DataHubSingleton = require("/data-hub/5/datahub-singleton.sjs");
+const datahub = DataHubSingleton.instance();
+const Artifacts = require('/data-hub/5/artifacts/core.sjs');
+
+var name;
+
+const flow = Artifacts.getArtifact("flow", name);
```

Comment:
I think you can get rid of a chunk of this code by moving the code in getFlowWithStepDetails.sjs into a library, which can go into the same folder as this endpoint - e.g. flow-lib.sjs (I think the impl/flow.sjs that has all the code for running a step should really be called something else besides "flow.sjs" at some point, but that's a separate matter). 

You can have a function of "getFlowsWithStepDetails(flowName)", where flowName is an optional parameter. If it exists, you call getArtifact, else getArtifacts. Then the same code can be used for constructing the flow with all the steps. And then in this endpoint, you take the result and add the job data for the single flow. 

That'll cut down on the duplication, and I think it'll be good to start a pattern of having a library in a DS folder for reusable code. That library could always move to a non-DS-specific location in the future, but I think the DS folder is a fine home for it for now.